bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/InterleavedLedgerStorage.java

```diff
@@ -98,11 +98,32 @@ public void initialize(ServerConfiguration conf,
                            Checkpointer checkpointer,
                            StatsLogger statsLogger)
             throws IOException {
+        initializeWithEntryLogListener(
+            conf,
+            ledgerManager,
+            ledgerDirsManager,
+            indexDirsManager,
+            stateManager,
+            checkpointSource,
+            checkpointer,
+            this,
+            statsLogger);
+    }
+
+    void initializeWithEntryLogListener(ServerConfiguration conf,
```

Comment:
instead of adding a new initialize method, you should consider changing the signature of the existing initialize method, which takes EntryLogListener argument. In ILS initialize method it will check if it receives the listener, if it is not null use it otherwise use ‘this’.